pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -54,596 +31,6 @@ int socket__classifier_dbs(struct __sk_buff *skb) {
-    }
-
-    if (!read_conn_tuple_partial(&t, st->sk, pid_tgid, CONN_TYPE_UDP)) {
-        log_debug("ERR(kretprobe/udp_recvmsg): error reading conn tuple, pid_tgid=%d\n", pid_tgid);
-        bpf_map_delete_elem(udp_sock_map, &pid_tgid);
-        return 0;
-    }
-    bpf_map_delete_elem(udp_sock_map, &pid_tgid);
-
-    log_debug("kretprobe/udp_recvmsg: pid_tgid: %d, return: %d\n", pid_tgid, copied);
-    // segment count is not currently enabled on prebuilt.
-    // to enable, change PACKET_COUNT_NONE => PACKET_COUNT_INCREMENT
-    handle_message(&t, 0, copied, CONN_DIRECTION_UNKNOWN, 0, 1, PACKET_COUNT_NONE, st->sk);
-
-    return 0;
-}
-
-SEC("kretprobe/udp_recvmsg")
-int kretprobe__udp_recvmsg(struct pt_regs *ctx) {
-    u64 pid_tgid = bpf_get_current_pid_tgid();
-    bpf_map_delete_elem(&udp_recv_sock, &pid_tgid);
-    return 0;
-}
-
-SEC("kretprobe/udpv6_recvmsg")
-int kretprobe__udpv6_recvmsg(struct pt_regs *ctx) {
-    u64 pid_tgid = bpf_get_current_pid_tgid();
-    bpf_map_delete_elem(&udpv6_recv_sock, &pid_tgid);
-    return 0;
-}
-
-SEC("kretprobe/udp_recvmsg")
-int kretprobe__udp_recvmsg_pre_4_7_0(struct pt_regs *ctx) {
-    int copied = (int)PT_REGS_RC(ctx);
-    return handle_ret_udp_recvmsg(copied, &udp_recv_sock);
-}
-
-SEC("kretprobe/udpv6_recvmsg")
-int kretprobe__udpv6_recvmsg_pre_4_7_0(struct pt_regs *ctx) {
-    int copied = (int)PT_REGS_RC(ctx);
-    return handle_ret_udp_recvmsg(copied, &udpv6_recv_sock);
-}
-
-SEC("kprobe/skb_free_datagram_locked")
-int kprobe__skb_free_datagram_locked(struct pt_regs *ctx) {
-    u64 pid_tgid = bpf_get_current_pid_tgid();
-    udp_recv_sock_t *st = bpf_map_lookup_elem(&udp_recv_sock, &pid_tgid);
-    if (!st) { // no entry means a peek
-        return 0;
-    }
-
-    struct sock *sk = (struct sock *)PT_REGS_PARM1(ctx);
-    struct sk_buff *skb = (struct sk_buff *)PT_REGS_PARM2(ctx);
-    handle_skb_consume_udp(sk, skb, 0);
-    return 0;
-}
-
-SEC("kprobe/__skb_free_datagram_locked")
-int kprobe____skb_free_datagram_locked(struct pt_regs *ctx) {
-    u64 pid_tgid = bpf_get_current_pid_tgid();
-    udp_recv_sock_t *st = bpf_map_lookup_elem(&udp_recv_sock, &pid_tgid);
-    if (!st) { // no entry means a peek
-        return 0;
-    }
-
-    struct sock *sk = (struct sock *)PT_REGS_PARM1(ctx);
-    struct sk_buff *skb = (struct sk_buff *)PT_REGS_PARM2(ctx);
-    int len = (int)PT_REGS_PARM3(ctx);
-    handle_skb_consume_udp(sk, skb, len);
-    return 0;
-}
-
-SEC("kprobe/skb_consume_udp")
-int kprobe__skb_consume_udp(struct pt_regs *ctx) {
-    u64 pid_tgid = bpf_get_current_pid_tgid();
-    udp_recv_sock_t *st = bpf_map_lookup_elem(&udp_recv_sock, &pid_tgid);
-    if (!st) { // no entry means a peek
-        return 0;
-    }
-
-    struct sock *sk = (struct sock *)PT_REGS_PARM1(ctx);
-    struct sk_buff *skb = (struct sk_buff *)PT_REGS_PARM2(ctx);
-    int len = (int)PT_REGS_PARM3(ctx);
-    handle_skb_consume_udp(sk, skb, len);
-    return 0;
-}
-
 SEC("kprobe/tcp_retransmit_skb")
```

Comment:
why was this left here?

Reply:
We have a different implementation for prebuilt than runtime compiled or co-re for this particular kprobe.